src/createDependencyMap.ts

```diff
@@ -0,0 +1,32 @@
+import { PackageInfos } from "./types/PackageInfo";
+
+// @internal
+export interface DependencyMap {
+  dependencies: Map<string, Set<string>>;
+  dependents: Map<string, Set<string>>;
+}
+
+// @internal
+export function createDependencyMap(packages: PackageInfos) {
+  const map = {
+    dependencies: new Map<string, Set<string>>(),
+    dependents: new Map<string, Set<string>>(),
+  };
+
+  for (const [pkg, info] of Object.entries(packages)) {
+    const deps = Object.keys({ ...info.dependencies, ...info.devDependencies });
```

Comment:
Why are we spreading in devDeps here? This should be optional. We also looke like we are not checking for external deps versus repo deps, is that intentional, I would ideally want that behind an option flag as well... :)

Reply:
Peer dependencies would also be interesting behind an option flag.

Reply:
Should be easy to check for packages in PackagesInfo and exclude if not looking at external deps?